### DISCUSSION TITLE
`pnpm create ...` does not work with workspace packages

### DISCUSSION THREAD

**Discussion by chasewalden** [pnpm/pnpm#15912] [Ideas]
### Code to reproduce the issue:

Project:
```
project/
├─ packages/
│  ├─ apps/
│  │  ├─ ...
│  ├─ utils/
│  │  ├─ ...
├─ templates/
│  ├─ create-app/
│  ├─ create-util/
├─ .npmrc
├─ pnpm-workspace.yaml
├─ (package.json)
```

pnpm-workspace.yaml
```yaml
packages:
- packages/**
- templates/*
```

Run `pnpm create app` or `pnpm create util` (or `create @scope/app` , `create @scope/util` )

### Expected behavior:

The workspace package to be resolved as a starter-kit package, so the `bin` field in the `create-{app,util}/package.json` to be executed

### Actual behavior:

The `create-{app,util}` package is `dlx`ed from the npm registry, running the remote version (or failing to find a matching package)

### Additional information:

- `node -v`: 16.13.1
- OS: macOS

Unsure if this is a bug or a new feature, but it is at least counterintuitive. If this is currently expected behavior, I would like to propose introducing this as a feature. It would be nice in large monorepos to be able to use "local" starter-kits without requiring pushing it to a registry.

I tried adding a `package.json` to the monorepo root and adding the starter-kits as devDependencies a-la `pnpm add -WD create-app create-util` before running `pnpm create ...`, but causes the same result

**Comment by zkochan** [comment]
Does it work with yarn?

**Comment by chasewalden** [comment]
I just did a quick check with Yarn 1, and it appears to be a No.

I will try with Yarn 2 (with / without PnP), but I am not too familiar with Yarn 2 or PnP, so it might be some time

**Comment by zkochan** [comment]
I am not against such feature.

**Comment by chasewalden** [comment]
Also a No for Yarn 2

If the intended behavior is to mirror Yarn, then this is working as intended. However, it would definitely be a subtle nicety from a dev experience. 

> I am not against such feature.

I can try an implementation within the next few days when I have some spare time in the afternoon

**Comment by stevebeauge** [comment]
Hi,

Is there any plan to implement this feature?

I wish to be able to define local template specific to my repo.

thanks

**Comment by antl3x** [comment]
+1 for this

**Comment by altaywtf** [comment]
I'm working on a fix for this.

Written by an agent (Claude Code, Opus 5.5).

**Comment by altaywtf** [comment]
A fix is ready for review in https://github.com/pnpm/pnpm/pull/15845. CI is green.

Written by an agent (Claude Code, Opus 5.5).
